src/main/java/bio/terra/common/DaoUtils.java

```diff
@@ -87,7 +87,8 @@ public static String escapeFilter(String filter) {
 
   public static Array createSqlStringArray(Connection connection, List<String> list)
       throws SQLException {
-    return connection.createArrayOf("text", list.toArray());
+    Object[] array = Optional.ofNullable(list).orElse(List.of()).toArray();
```

Comment:
You can avoid creating an `Optional` here by using `Objects.requireNonNullElse`
```suggestion
    Object[] array = Objects.requireNonNullElse(list, List.of()).toArray();
```

Although personally I would avoid both forms and use `!= null` directly
```
    Object[] array = list != null ? list.toArray() : new Object[0];
```

Reply:
In a later commit (not yet pushed), I actually switched to `ListUtils.emptyIfNull` to get rid of the Optional, I agree that it was clunky.

A heads up that this PR is still very much in flight, you're welcome to look at it but the sand might shift underneath you while you do :)